tools/csv-merger/csv-merger.go

```
@@ -0,0 +1,375 @@
+					image = image_s[0]
+					name = image_s[1]
+				} else {
+					names := strings.Split(strings.Split(image, "@")[0], "/")
+					name = names[len(names)-1]
+				}
+				csvExtended.Spec.RelatedImages = append(
+					csvExtended.Spec.RelatedImages,
+					relatedImage{
+						Name: name,
+						Ref:  image,
+					})
+			}
+		}
+
+		for _, csvStr := range csvs {
+			if csvStr != "" {
+				csvBytes := []byte(csvStr)
+
+				csvStruct := &csvv1alpha1.ClusterServiceVersion{}
+
+				err := yaml.Unmarshal(csvBytes, csvStruct)
+				if err != nil {
+					panic(err)
+				}
+
+				installStrategyBase.DeploymentSpecs = append(installStrategyBase.DeploymentSpecs, csvStruct.Spec.InstallStrategy.StrategySpec.DeploymentSpecs...)
+				installStrategyBase.ClusterPermissions = append(installStrategyBase.ClusterPermissions, csvStruct.Spec.InstallStrategy.StrategySpec.ClusterPermissions...)
+				installStrategyBase.Permissions = append(installStrategyBase.Permissions, csvStruct.Spec.InstallStrategy.StrategySpec.Permissions...)
+
+				for _, owned := range csvStruct.Spec.CustomResourceDefinitions.Owned {
+					csvExtended.Spec.CustomResourceDefinitions.Owned = append(
+						csvExtended.Spec.CustomResourceDefinitions.Owned,
+						csvv1alpha1.CRDDescription{
+							Name:        owned.Name,
+							Version:     owned.Version,
+							Kind:        owned.Kind,
+							Description: owned.Description,
+							DisplayName: owned.DisplayName,
+						},
+					)
+				}
+
+				csv_base_alm_string := csvExtended.Annotations["alm-examples"]
+				csv_struct_alm_string := csvStruct.Annotations["alm-examples"]
+				var base_almcrs []interface{}
+				var struct_almcrs []interface{}
+				if err = json.Unmarshal([]byte(csv_base_alm_string), &base_almcrs); err != nil {
+					panic(err)
+				}
+				if err = json.Unmarshal([]byte(csv_struct_alm_string), &struct_almcrs); err == nil {
+					//panic(err)
+					for _, cr := range struct_almcrs {
+						base_almcrs = append(
```

base_almcrs = extended["alm-examples"] + struct["alm-examples"] // ignoring errors in the latter